packages/core/src/data-grid/data-grid-lib.ts

```diff
@@ -377,30 +378,56 @@ function drawCheckbox(
 
     const hovered = Math.abs(hoverX - width / 2) < 10 && Math.abs(hoverY - height / 2) < 10;
 
-    if (checked) {
-        ctx.beginPath();
-        roundedRect(ctx, centerX - 9, centerY - 9, 18, 18, 4);
+    switch (checked) {
+        case true: {
+            ctx.beginPath();
+            roundedRect(ctx, centerX - 9, centerY - 9, 18, 18, 4);
 
-        ctx.fillStyle = highlighted ? theme.accentColor : theme.textLight;
-        ctx.fill();
+            ctx.fillStyle = highlighted ? theme.accentColor : theme.textMedium;
```

Comment:
Why medium?

Reply:
2 folded:
- I needed a pale color for the indeterminate state that was easy to distinguish from the checked state.
- This bumps contrast and still looks nice